File: Archive#02-201-300/283-Move Zeroes/leetcode283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        non_zeros = 0
        for i in range(len(nums)):
            if nums[i]:
                nums[i], nums[non_zeros] = nums[non_zeros], nums[i]
                non_zeros += 1

File: Archive#02-201-300/283-Move Zeroes/leetcode283.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
  non_zeros = 0
  (0...nums.length).each {|i|
    if nums[i] != 0 then
      nums[i], nums[non_zeros] = nums[non_zeros], nums[i]
      non_zeros += 1
    end
  }
end